### PR TITLE
Re-enable tests for macOS

### PR DESCRIPTION
**Pull request by m-kuhn**
This enables travis testing for macos.

# TODO

 * unblacklist all the failing tests
 * possibly create (and upload) homebrew bottles of QGIS 3 binaries

# Cache warming

Because the build takes too long with a cold cache, some magic has been introduced. If the build takes too long, it will be killed early enough to allow the ccache to be uploaded. In this case, re-starting the build will continue the build with an updated cache which hopefully reduces the build time enough to complete (or it will need to be restarted yet once again).

### CODE REVIEW

File: ci/travis/macos/before_install.sh [hunk 1]
```diff
@@ -13,29 +13,51 @@
...
+    ccache \
+    spatialindex \
+    fastcgi \
+    qtkeychain \
+    gnu-sed
 
-# Needed for Processing
-pip install psycopg2 numpy nose2 pyyaml mock future
+brew link bison --force
+brew link flex --force
```
Comment (dakcarto) [inline ci/travis/macos/before_install.sh]: Not sure forcing the install of these are still necessary if using the CMAKE_PREFIX_PATH prefixes in `install.sh`

File: ci/travis/macos/before_install.sh [hunk 2]
```diff
@@ -13,29 +13,51 @@
...
+
+
+# Fix qscintilla typo
+wget https://gist.githubusercontent.com/m-kuhn/f70e4b160dd7b18eb8d637ed2a75df6d/raw/6eb8a0c8601ec52ad9ad41c01d0fad68fe967aa1/qsci.patch
+patch -p1 /usr/local/share/sip/QSci/qscilexer.sip qsci.patch
```
Comment (dakcarto) [inline ci/travis/macos/before_install.sh]: I will look at adding this to the upstream Homebrew-core formula.

**Comment by dakcarto** [comment]
@m-kuhn Awesome! Thanks for doing this.

File: ci/travis/macos/before_install.sh (hunk 2 above)
Comment (m-kuhn) [inline ci/travis/macos/before_install.sh]: cool, was also looking but was interrupted... patch is working but with warnings.

File: ci/travis/macos/before_install.sh (hunk 1 above)
Comment (m-kuhn) [inline ci/travis/macos/before_install.sh]: Yes maybe not. It's probably a leftover from before the PATH was correctly set u

**Pull request merged**